Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/WebAppWrapper.java

```diff
@@ -132,4 +134,44 @@ private WebApp getWebApp() {
             throw new RuntimeException("Failed to get webapp instance");
         }
     }
+
+    @Override
+    public void deploy(DeployType type, File file) {
+        this.getWebApp().deploy(type, file);
+    }
+
+    @Override
+    public Completable deployAsync(DeployType type, File file) {
```

Comment:
where is the call to `deployAsync`?

Reply:
nowhere. the existing WebAppWrapper implements com.microsoft.azure.management.appservice.WebApp. SDK changed to add this method in its interface on one deploy version. This is forced to implement.